Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -186,16 +186,19 @@ public ArraySegment<byte> ReadBytesSegment(int count)
         // null support, see NetworkWriter
         public byte[] ReadBytesAndSize()
         {
-            if (ReadBoolean())
-            {
-                return ReadBytes(checked((int)ReadPackedUInt32()));
-            }
-            return null;
+            // count = 0 means the array was null
+            // otherwise count -1 is the length of the array 
+            uint count = ReadPackedUInt32();
```

Comment:
why don't we use regular PackedInt here though?
for empty[] we would write -1, which would use 5 bytes if we use PackedUint

Reply:
We can,  and in fact,  that is how arrays are done in the weaver (until I change it).

But this is better,  consider an array of length 120,    it would take 2 bytes with packet index, but only 1 byte with length + 1.     Yes,  it is a micro optimization that does not really matter,  but,  in my mind this is simpler.